pg_service_parser/core/connection_model.py

```diff
@@ -32,7 +32,7 @@ def data(self, index, role=Qt.ItemDataRole.DisplayRole):
             if index.column() == self.KEY_COL:
                 return key
             elif index.column() == self.VALUE_COL:
-                return self.__model_data[key].uri()
+                return QgsDataSourceUri.removePassword(self.__model_data[key].uri(), True)
```

Comment:
Nice method, I was not aware of it.

However, did you try it out?
For me an URI like `service='system-nyc' user='abc' password='def'` remains untouched after running:

    QgsDataSourceUri.removePassword(uri, True)

It might be an upstream issue.

Reply:
I tried it with my service file and it worked. I will test further

Reply:
I just tested it on a project inside the Python console. It works as expected :

```python
QgsDataSourceUri.removePassword(iface.activeLayer().dataProvider().uri().uri(), True)
```

Reply:
hum, your example does not work indeed....

Reply:
I will add a `replace` to remove the password. Something like

```python
uri.uri().replace(f"password='{uri.password()}'", "password='XXXXXXX'")
```